keps/sig-api-machinery/2876-crd-validation-expression-language/README.md

```diff
@@ -515,18 +565,20 @@ The initial changes made in the type integration will be:
   - We couldn't have both: (a) equality of string keys, which implies a canonical ordering of key fields, and (b) ability for developers to successfully lookup a map entry regardless of how they order the keys in the string representation
 
 
-So instead of treating "associative lists" as maps in CEL, we will continue to treat them as lists, but override equality to ignore object order (i.e. use map equality semantics) and introduce utility functions to make the list representation easy to use in CEL. E.g. instead of:
+So instead of treating "associative lists" as maps in CEL, we will continue to treat them as lists, but override equality to ignore object order (i.e. use map equality semantics).
+
+Looking up entiries by keys is available primarily via the `exists_one` and `filter` macros. Examples:
 
 ```
-associativeList.filter(e, e.key1 == 'a' && e.key2 == 'b').all(e, e.val == 100)
-```
+// exists_one() and exists() behave similarly if all map keys are checked, but exists_one() has slightly stricter
+// semantics, which make it preferable
```

Comment:
`exists_one` is good if there must be exactly one entry that matches, but otherwise, when evaluating predicates over dynamic data (such as map fields) the `exists` and `all` might be easier to wield successfully. Maybe it's worth covering when to use these macros as well?

That said, I would omit discussing `filter` and `map` if you can since the removed example could have been expressed as `associativeList.exists(e, e.key1 == 'a' && e.key2 == 'b' && e.val == 100)`

Reply:
Thanks, I've cleaned this up to use `exists` (and refer to the go spec for some of the more subtle error handling differences between `exists` and `exists_one`), added some `all` examples and drop the reference to `filter`.